caffe2/operators/experimental/c10/cpu/concat_cpu.cc

```diff
@@ -13,46 +13,46 @@ namespace caffe2 {
 namespace {
 template <class DataType, class Context>
 void concat_op_cpu_impl(
-    at::ArrayRef<C10Tensor> inputs,
-    const C10Tensor& output_,
-    const C10Tensor& split_,
+    intrusive_ptr<ivalue::TensorList> inputs,
```

Comment:
This looks weird to me.

Based on our discussion, we said there were two plausible ways operators could be written:
* Taking a `Stack` as argument, in which case they feed in and out ivalues
* The "native" format, which "looks something like how ATen functions are written."

What we have here is some weird bastard format. It isn't Stack. But it isn't native either, because you're taking in an `ivalue::TensorList`. No one should actually write a kernel this way; they should just take `ArrayRef<Tensor>`. What's the plan?

Reply:
Can you remind me where the test for these experimental ops lives?

Reply:
good point, I should change this to `ArrayRef<Tensor>`. There are currently only tests for layer_norm, the other ops are only there to make sure certain use cases still compile (e.g. operators with/without state, operators with/without attributes, operators with/without TensorList as input, single/multiple outputs, ...), but they don't have test cases, so they only check it's syntactically possible.